src/sql/src/plan/statement/dml.rs

```diff
@@ -486,7 +483,7 @@ pub fn plan_query(
     query: Query<Aug>,
     params: &Params,
     lifetime: QueryLifetime,
-) -> Result<query::PlannedRootQuery<MirRelationExpr>, PlanError> {
+) -> Result<query::PlannedRootQuery<HirRelationExpr>, PlanError> {
```

Comment:
I am very unsure about the change here. If it is safe to make, the fn comment needs to be changed. But I'm not at all certain about if it's safe to change, and am sadly not familiar enough with planning to know if this matters or not. @benesch do you happen to remember if this needs to be a Mir for a specific reason? Is it wrong or risky to decorrelate at a later time?

Reply:
I'm just now realizing that there is a dedicated `plan_root_query` which does exactly what I want (planning a `SELECT` query without decorrelation). I asked @benesch [the same question ](https://github.com/MaterializeInc/materialize/pull/24581#issuecomment-1903775294) in #24581 (one answer is sufficient).

Reply:
My gut feeling is that relying on something that `MirRelationExpr` gives you, but `HirRelationExpr` doesn't to complete the planning process is a smell, and I'm curious to learn what this thing might be.

Reply:
It's `MirRelationExpr` only because previously that was the contract of the SQL planner: it takes in an AST and produces MIR. No fundamental reason that we can't change that contract to be AST -> HIR, like Alexander's done in #24581.